apps/designer/src/components/FloorToggleTab.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from "react";
+import cameraControlsStore from "../globalStore/Navigation-Store/cameraControlsStore";
+import { cameraAngles } from "../../public/cameraPropConstants.json";
+
+const FloorToggleTab = () => {
+  const floor = cameraControlsStore((state) => state.floor);
+  const updateTarget = cameraControlsStore((state) => state.updateCameraTarget);
+  const updatePosition = cameraControlsStore(
+    (state) => state.updateCameraPosition
+  );
+  const changeFloor = cameraControlsStore((state) => state.changeFloor);
+
+  const updateFloor = (idx: number) => {
+    changeFloor(idx);
+    updatePosition(cameraAngles[idx].cameraPosition);
+    updateTarget(cameraAngles[idx].cameraTarget);
+  };
+  function activeFloorTab(idx: number) {
```

Comment:
1. make return type explicit
2. name should indicate that this is a conditional style